### DISCUSSION TITLE
Repository - Select columns

### DISCUSSION THREAD

**Discussion by mrgiacomini** [jhipster/jhipster-dotnetcore#873] [General]
Hi

How can I select specific columns from my entity?  In QueryHandler, I can only filter (similar to Where in SQL), but I only need to get the specific columns of the table

**Comment by ivanmonteiro** [comment]
> Hi
> 
> How can I select specific columns from my entity? In QueryHandler, I can only filter (similar to Where in SQL), but I only need to get the specific columns of the table

I think you want to use [projections](https://entityframeworkcore.com/querying-data-projection) to fetch only the properties/columns you need.
Unfortunately the current implementation of GenericRepository and QueryHelper does not have the appropriate extension methods to use projections.

But it would not be hard to add it, here is one example:
```
public async Task<TResult> GetById<TResult>(int id, Expression<Func<TEntity, TResult>> selector)
{
   return await this.GetById(id)
       .Select(selector)
       .SingleOrDefaultAsync();
}
```

Altough I don't have the time to implement it, if you want to contribute I might provide assistance.
